mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,311 @@
+        "spark.mlflow.pysparkml.autolog.logModelAllowlistFile", None
+    )
+    if allowlist_file:
+        try:
+            return _read_log_model_allowlist_from_file(allowlist_file)
+        except Exception:
+            # fallback to built-in allowlist file
+            _logger.exception(
+                "Reading from custom log_models allowlist file "
+                + "%s failed, fallback to built-in allowlist file.",
+                allowlist_file,
+            )
+            return _read_log_model_allowlist_from_file(builtin_allowlist_file)
+    else:
+        return _read_log_model_allowlist_from_file(builtin_allowlist_file)
+
+
+# When autolog called, `_log_model_allowlist` will be initialized.
+_log_model_allowlist = None
+
+
+def _get_warning_msg_for_skip_log_model(model):
+    return (
+        f"Model {model.uid} will not be autologged because it is not allowlisted or or because "
+        "one or more of its nested models are not allowlisted. Call mlflow.spark.log_model() "
+        "to explicitly log the model, or specify a custom allowlist via the "
+        "spark.mlflow.pysparkml.autolog.logModelAllowlistFile Spark conf "
+        "(see mlflow.pyspark.ml.autolog docs for more info)."
+    )
+
+
+def _should_log_model(spark_model):
+    # TODO: Handle PipelineModel/CrossValidatorModel/TrainValidationSplitModel
+    class_name = _get_fully_qualified_class_name(spark_model)
+    if class_name in _log_model_allowlist:
+        if class_name == "pyspark.ml.classification.OneVsRestModel":
+            return _should_log_model(spark_model.models[0])
+        else:
+            return True
+    else:
+        return False
+
+
+def _get_estimator_info_tags(estimator):
+    """
+    :return: A dictionary of MLflow run tag keys and values
+             describing the specified estimator.
+    """
+    return {
+        "estimator_name": estimator.__class__.__name__,
+        "estimator_class": _get_fully_qualified_class_name(estimator),
+    }
+
+
+def _get_instance_param_map(instance):
+    from pyspark.ml.param import Params
+    from pyspark.ml.tuning import CrossValidator, TrainValidationSplit
+
+    param_map = {
+        param.name: instance.getOrDefault(param)
+        for param in instance.params
+        if instance.isDefined(param)
+    }
+    expanded_param_map = {}
+    for k, v in param_map.items():
+        if isinstance(v, Params):
+            # handle the case param value type inherits `pyspark.ml.param.Params`
+            # e.g. param like `OneVsRest.classifier`/`CrossValidator.estimator`
+            expanded_param_map[k] = v.uid
+            internal_param_map = _get_instance_param_map(v)
+            for ik, iv in internal_param_map.items():
+                expanded_param_map[f"{v.uid}.{ik}"] = iv
+        elif k == "estimatorParamMaps" and isinstance(
+            instance, (CrossValidator, TrainValidationSplit)
+        ):
+            # TODO: Log `estimatorParamMaps` as JSON artifacts.
```

Comment:
Will also do this in follow-up tuning support PR , this PR discussions are too long.